packages/admin/cms-admin/src/generator/generate-form.ts

```diff
@@ -0,0 +1,347 @@
+    import { FormApi } from "final-form";
+    import { filter } from "graphql-anywhere";
+    import isEqual from "lodash.isequal";
+    import React from "react";
+    import { FormattedMessage } from "react-intl";
+    import { useContentScope } from "@src/common/ContentScopeProvider";
+    import { create${entityName}Mutation, ${instanceEntityName}CheckForChangesQuery, ${instanceEntityName}FormFragment, ${instanceEntityName}FormQuery, update${entityName}Mutation } from "./${entityName}Form.gql";
+    import {
+        GQL${entityName}FormCheckForChangesQuery,
+        GQL${entityName}FormCheckForChangesQueryVariables,
+        GQL${entityName}FormCreateMutation,
+        GQL${entityName}FormCreateMutationVariables,
+        GQL${entityName}FormFragment,
+        GQL${entityName}FormUpdateMutation,
+        GQL${entityName}FormUpdateMutationVariables,
+        GQL${entityName}FormQuery,
+        GQL${entityName}FormQueryVariables,
+    } from "./${entityName}Form.gql.generated";
+    ${Object.entries(rootBlocks)
+        .map(([rootBlockKey, rootBlock]) => `import { ${rootBlock.name} } from "${rootBlock.import}";`)
+        .join("\n")}
+    interface FormProps {
+        id?: string;
+    }
+    
+    ${
+        Object.keys(rootBlocks).length > 0
+            ? `const rootBlocks = {
+                ${Object.entries(rootBlocks).map(([rootBlockKey, rootBlock]) => `${rootBlockKey}: ${rootBlock.name}`)}
+                };`
+            : ""
+    }
+    
+    type FormValues = ${
+        numberFields.length > 0
+            ? `Omit<GQL${entityName}FormFragment, ${numberFields.map((field) => `"${field.name}"`).join(", ")}>`
+            : `GQL${entityName}FormFragment`
+    } & {
+        ${numberFields.map((field) => `${field.name}: string;`).join("\n")}
+        ${Object.keys(rootBlocks)
+            .map((rootBlockKey) => `${rootBlockKey}: BlockState<typeof rootBlocks.${rootBlockKey}>;`)
+            .join("\n")}
+    };
+    
+    export function ${entityName}Form({ id }: FormProps): React.ReactElement {
+        const stackApi = useStackApi();
+        const client = useApolloClient();
+        const mode = id ? "edit" : "add";
+        const formApiRef = useFormApiRef<FormValues>();
+        const stackSwitchApi = useStackSwitchApi();
+        ${hasScope ? `const { scope } = useContentScope();` : ""}
+    
+        const { data, error, loading, refetch } = useQuery<GQL${entityName}FormQuery, GQL${entityName}FormQueryVariables>(${instanceEntityName}FormQuery,
+            id ? { variables: { id } } : { skip: true },
+        );
+    
+        const initialValues: Partial<FormValues> = data?.${instanceEntityName}
+            ? {
+                  ...filter<GQL${entityName}FormFragment>(${instanceEntityName}FormFragment, data.${instanceEntityName}),
+                  ${numberFields.map((field) => `${field.name}: String(data.${instanceEntityName}.${field.name}),`).join("\n")}
```

Comment:
```suggestion
                  ${numberFields.length > 0 ? numberFields.map((field) => `${field.name}: String(data.${instanceEntityName}.${field.name}),`).join("\n"): ""}
```

This causes the empty line